vtysh/vtysh_ovsdb_router_context.c

```diff
@@ -48,147 +218,91 @@
-
-            if (nbr_table->shutdown)
-                vtysh_ovsdb_cli_print(p_msg, "%4s %s %s %s", "", "neighbor",
-                                      neighbor,"shutdown");
-
-            if (nbr_table->n_advertisement_interval)
-                vtysh_ovsdb_cli_print(p_msg,"%4s %s %s %s %d", "", "neighbor",
-                                      neighbor,
-                                      "advertisement-interval", *(nbr_table->
-                                      advertisement_interval));
-
-            if (nbr_table->n_timers > 0)
-                vtysh_ovsdb_cli_print(p_msg, "%4s %s %s %s %d %d","","neighbor",
-                                      neighbor, "timers",
-                                      nbr_table->value_timers[1],
-                                      nbr_table->value_timers[0]);
-
-            if (nbr_table->n_bfd_fallover_enable > 0)
-                vtysh_ovsdb_cli_print(p_msg, "%4s %s %s %s %s","","neighbor",
-                                      neighbor, "fall-over", "bfd");
-
-            i=0;
-            while (i < nbr_table->n_route_maps) {
-                vtysh_ovsdb_cli_print(p_msg, "%4s %s %s %s %s %s", "",
-                                      "neighbor", neighbor,
-                                      "route-map", nbr_table->
-                                      value_route_maps[i]->name,
-                                      nbr_table->key_route_maps[i]);
-                i++;
-            }
+            nbr_table  = bgp_router_context->value_bgp_neighbors[n_neighbors];
+            neighbor   = bgp_router_context->key_bgp_neighbors[n_neighbors];
+            pgroup_ptr = NULL;
 
-            i=0;
-            while (i < nbr_table->n_prefix_lists) {
-                vtysh_ovsdb_cli_print(p_msg, "%4s %s %s %s %s %s", "",
-                                      "neighbor", neighbor,
-                                      "prefix-list", nbr_table->value_prefix_lists[i]->name,
-                                      nbr_table->key_prefix_lists[i]);
-                i++;
-            }
+          /* Neighbor individual peer commands only. */
+            if (*(nbr_table->is_peer_group)) continue;
 
-            i=0;
-            while (i < nbr_table->n_aspath_filters) {
-                vtysh_ovsdb_cli_print(p_msg, "%4s %s %s %s %s %s", "",
-                                      "neighbor", neighbor,
-                                      "filter-list", nbr_table->value_aspath_filters[i]->name,
-                                      nbr_table->key_aspath_filters[i]);
-                i++;
+          /* Store pointer to peer-group ref of neighbor for quick access. */
+            if (nbr_table->bgp_peer_group) {
+                for (k = 0; k < n_pgroups; k++) {
+                    if (nbr_table->
+                        bgp_peer_group == pgroup_tables_ptr[k]) {
+                            pgroup_ptr = pgroup_tables_ptr[k];
+                    }
+                }
```

Comment:
Can we avoid using of extraarray "pgroup_tables_ptr" at all, if condition (267th line) would be like:
if (nbr_table->bgp_peer_group) {
pgroup_ptr = nbr_table->bgp_peer_group;
} else {
pgroup_ptr = NULL;
}
As I understand we have the same result.